Examples/DpdkBridge/main.cpp

```diff
@@ -0,0 +1,336 @@
+
+
+/**
+ * Print application version
+ */
+void printAppVersion()
+{
+	printf("%s %s\n", AppName::get().c_str(), getPcapPlusPlusVersionFull().c_str());
+	printf("Built: %s\n", getBuildDateTime().c_str());
+	printf("Built from: %s\n", getGitInfo().c_str());
+	exit(0);
+}
+
+
+/**
+ * Print to console all available DPDK ports. Used by the -l switch
+ */
+void listDpdkPorts()
+{
+	CoreMask coreMaskToUse = getCoreMaskForAllMachineCores();
+
+	// initialize DPDK
+	if (!DpdkDeviceList::initDpdk(coreMaskToUse, DEFAULT_MBUF_POOL_SIZE))
+	{
+		EXIT_WITH_ERROR("couldn't initialize DPDK");
+	}
+
+	printf("DPDK port list:\n");
+
+	// go over all available DPDK devices and print info for each one
+	vector<DpdkDevice*> deviceList = DpdkDeviceList::getInstance().getDpdkDeviceList();
+	for (vector<DpdkDevice*>::iterator iter = deviceList.begin(); iter != deviceList.end(); iter++)
+	{
+		DpdkDevice* dev = *iter;
+		printf("    Port #%d: MAC address='%s'; PCI address='%s'; PMD='%s'\n",
+				dev->getDeviceId(),
+				dev->getMacAddress().toString().c_str(),
+				dev->getPciAddress().toString().c_str(),
+				dev->getPMDName().c_str());
+	}
+}
+
+struct DpdkBridgeArgs
+{
+	bool shouldStop;
+	std::vector<DpdkWorkerThread*>* workerThreadsVector;
+
+	DpdkBridgeArgs() : shouldStop(false), workerThreadsVector(NULL) {}
+};
+
+/**
+ * The callback to be called when application is terminated by ctrl-c. Do cleanup and print summary stats
+ */
+void onApplicationInterrupted(void* cookie)
+{
+	DpdkBridgeArgs* args = (DpdkBridgeArgs*)cookie;
+
+	printf("\n\nApplication stopped\n");
+
+	// stop worker threads
+	DpdkDeviceList::getInstance().stopDpdkWorkerThreads();
+
+	// create table printer
+	std::vector<std::string> columnNames;
+	std::vector<int> columnWidths;
+	PacketStats::getStatsColumns(columnNames, columnWidths);
+	TablePrinter printer(columnNames, columnWidths);
+
+	// print final stats for every worker thread plus sum of all threads and free worker threads memory
+	PacketStats aggregatedStats;
+	for (std::vector<DpdkWorkerThread*>::iterator iter = args->workerThreadsVector->begin(); iter != args->workerThreadsVector->end(); iter++)
+	{
+		AppWorkerThread* thread = (AppWorkerThread*)(*iter);
+		PacketStats threadStats = thread->getStats();
+		aggregatedStats.collectStats(threadStats);
+		printer.printRow(threadStats.getStatValuesAsString("|"), '|');
+		delete thread;
+	}
+
+	printer.printSeparator();
+	printer.printRow(aggregatedStats.getStatValuesAsString("|"), '|');
+
+	args->shouldStop = true;
+}
+
+
+/**
+ * main method of the application. Responsible for parsing user args, preparing worker thread configuration, creating the worker threads and activate them.
+ * At program termination worker threads are stopped, statistics are collected from them and printed to console
+ */
+int main(int argc, char* argv[])
+{
+	AppName::init(argc, argv);
+
+	std::vector<int> dpdkPortVec;
+
+	CoreMask coreMaskToUse = getCoreMaskForAllMachineCores();
+
+	int sendPacketsToPort = -1;
```

Comment:
This is not used anywhere, please remove it

Reply:
Agreed, I removed it